src/components/DateTimePicker/index.js

```diff
@@ -0,0 +1,217 @@
+    /** Sets the date for the DateTimePicker programmatically. */
+    value: PropTypes.instanceOf(Date),
+    /** The beginning of a range of valid dates. The range includes the startDate.
+     * The default value is current date - 100 years. */
+    minDate: PropTypes.instanceOf(Date),
+    /** The ending of a range of valid dates. The range includes the endDate.
+     * The default value is current date + 100 years. */
+    maxDate: PropTypes.instanceOf(Date),
+    /**  The date time format style to display in the input field.
+     * Valid values are small, medium, and large. */
+    formatStyle: PropTypes.oneOf(['small', 'medium', 'large']),
+    /** The name of the DateTimePicker. */
+    name: PropTypes.string,
+    /** Text label for the DateTimePicker. */
+    label: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    /** A boolean to hide the DateTimePicker label. */
+    hideLabel: PropTypes.bool,
+    /** Text that is displayed when the DateTimePicker is empty,
+     * to prompt the user for a valid entry. */
+    placeholder: PropTypes.string,
+    /** Specifies that the DateTimePicker field must be filled out before submitting the form.
+     * This value defaults to false. */
+    required: PropTypes.bool,
+    /** Shows the help message below the DateTimePicker. */
+    bottomHelpText: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** Specifies that the DateTimePicker text will be centered. This value defaults to false. */
+    isCentered: PropTypes.bool,
+    /** Specifies that the DateTimePicker must be filled out before submitting the form. */
+    error: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
+    /** Specifies that the DateTimePicker is read-only. This value defaults to false. */
+    readOnly: PropTypes.bool,
+    /** Specifies that the DateTimePicker element should be disabled.
+     * This value defaults to false. */
+    disabled: PropTypes.bool,
+    /** Specifies the tab order of an element (when the tab button is used for navigating). */
+    tabIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
+    /** The action triggered when a value attribute changes. */
+    onChange: PropTypes.func,
+    /** The action triggered when the element is clicked. */
+    onClick: PropTypes.func,
+    /** The action triggered when the element receives focus. */
+    onFocus: PropTypes.func,
+    /** The action triggered when the element releases focus. */
+    onBlur: PropTypes.func,
+    /** The id of the outer element. */
+    id: PropTypes.string,
+    /** A CSS class for the outer element, in addition to the component's base classes. */
+    className: PropTypes.string,
+    /** An object with custom style applied to the outer element. */
+    style: PropTypes.object,
+    /** Text label for the OK button in the modal dialog. */
+    okLabel: PropTypes.string,
+    /** Text label for the CANCEL button in the modal dialog. */
+    cancelLabel: PropTypes.string,
```

Comment:
The `cancelLabel` prop should be `oneOfType([PropTypes.string, PropTypes.node]),`